### PR TITLE
Fixed a parquet python module failure when s3 extension is not included

### LINKED ISSUE #5091: parquet.py depends on s3 being available
parquet.py should not have a hard dependency on s3; this means that it should not import s3.

```
2024-01-31T16:29:10.819Z | r-Scheduler-Serial-1 | ERROR | .c.ConsoleServiceGrpcImpl | Error running script: java.lang.RuntimeException: Error in Python interpreter:
Type: <class 'ValueError'>
Value: Java class 'io.deephaven.extensions.s3.AwsCredentials' not found
Line: 15
Namespace: <module>
File: /tmp/dh/lib64/python3.12/site-packages/deephaven/experimental/s3.py
Traceback (most recent call last):
  File "<string>", line -1, in <module>
  File "/tmp/dh/lib64/python3.12/site-packages/deephaven/parquet.py", line -1, in <module>
  File "/tmp/dh/lib64/python3.12/site-packages/deephaven/experimental/s3.py", line -1, in <module>
```

### PR DESCRIPTION
**Pull request by malhotrashivam**
Introduced in #4972 
Closes #5091

### CODE REVIEW

File: py/server/deephaven/experimental/s3.py [hunk 1]
```diff
@@ -26,7 +31,7 @@ class S3Instructions(JObjectWrapper):
     S3Instructions provides specialized instructions for reading from AWS S3.
     """
 
-    j_object_type = _JS3Instructions
+    j_object_type = _JS3Instructions if _JS3Instructions is not None else type(None)
```
Comment (devinrsmith) [inline py/server/deephaven/experimental/s3.py:34]: `_JS3Instructions or type(None)` is more idiomatic.

Comment (jmao-denver) [reply py/server/deephaven/experimental/s3.py:34]: +1

File: py/server/deephaven/experimental/s3.py [hunk 2]
```diff
@@ -73,6 +78,10 @@ def __init__(self,
             DHError: If unable to build the instructions object.
         """
 
+        if _JS3Instructions is None or _JAwsCredentials is None:
```
Comment (jmao-denver) [inline py/server/deephaven/experimental/s3.py:81]:
```suggestion
        if not _JS3Instructions or not _JAwsCredentials:
```

**Review by jmao-denver — CHANGES_REQUESTED** [review]

**Review by jmao-denver — APPROVED** [review]
LGTM

**Review by devinrsmith — APPROVED** [review]

**Pull request merged**